sys/kern/imgact_elf.c

```diff
@@ -510,6 +515,56 @@ __elfN(phdr_in_zero_page)(const Elf_Ehdr *hdr)
 	    (u_int)hdr->e_phentsize * hdr->e_phnum <= PAGE_SIZE - hdr->e_phoff);
 }
 
+static const Elf_Phdr *
+__elfN(map_phdr)(struct image_params *imgp, const Elf_Ehdr *hdr, bool complain)
+{
+	struct thread *td;
+	void *buf;
+	size_t len;
+	int error;
+
+	if (__elfN(phdr_in_zero_page)(hdr))
+		return ((const Elf_Phdr *)(imgp->image_header + hdr->e_phoff));
+
+	len = hdr->e_phnum * hdr->e_phentsize;
+	if (len > __elfN(max_phdr_len)) {
+		if (complain)
+			uprintf("Program headers too large\n");
```

Comment:
While you certainly can read the size from the binary, printing it might make life easier